src/js/components/PodInstancesTable.js

```diff
@@ -0,0 +1,337 @@
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'mem',
+        render: this.renderColumnResource,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'updated',
+        render: this.renderColumnUpdated,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'version',
+        render: this.renderColumnVersion,
+        sortable: true
+      }
+    ];
+  }
+
+  getContainersWithResources(podSpec, instance) {
+    let resourcesSum = {
+      cpus: 0, mem: 0, disk: 0, gpus: 0
+    };
+
+    let containers = instance.getContainers();
+    let children = containers.map(function (container) {
+      let containerSpec = podSpec.getContainerSpec(container.name);
+      Object.keys(containerSpec.resources).forEach(function (key) {
+        resourcesSum[key] += containerSpec.resources[key];
+      });
+
+      let addressComponents = container.endpoints.map(function (endpoint, i) {
+        return (
+          <a className="text-muted"
+            href={`http://${instance.getAgentAddress()}:${endpoint.allocatedHostPort}`}
+            key={i}
+            target="_blank"
+            title="Open in a new window">
+            {':' + endpoint.allocatedHostPort}
+          </a>
+        );
+      });
+
+      return {
+        id: container.getId(),
+        name: container.getName(),
+        status: container.getContainerStatus(),
+        address: addressComponents,
+        cpus: containerSpec.resources.cpus,
+        mem: containerSpec.resources.mem,
+        updated: container.getLastUpdated(),
+        version: ''
+      };
+    });
+
+    return {children, resourcesSum};
+  }
+
+  getTableDataFor(instances) {
+    let podSpec = this.props.pod.getSpec();
+
+    return instances.getItems().map((instance) => {
+      let {children, resourcesSum} = this.getContainersWithResources(
+        podSpec, instance
+      );
+
+      return {
+        id: instance.getId(),
+        name: instance.getName(),
+        address: instance.getAgentAddress(),
+        cpus: resourcesSum.cpus,
+        mem: resourcesSum.mem,
+        updated: instance.getLastUpdated(),
+        status: instance.getInstanceStatus(),
+        version: podSpec.getVersion(),
+        children
+      };
+    });
+
+  }
+
+  getInstanceFilterStatus(instance) {
+    let status = instance.getInstanceStatus();
+    switch (status) {
+      case PodInstanceStatus.STAGED:
+        return 'staged';
+
+      case PodInstanceStatus.HEALTHY:
+      case PodInstanceStatus.UNHEALTHY:
+      case PodInstanceStatus.RUNNING:
+        return 'active';
+
+      case PodInstanceStatus.KILLED:
+        return 'completed';
+
+      default:
+        return '';
+    }
+  }
+
+  renderWithClickHandler(rowOptions, className, content) {
```

Comment:
#nit Could we move `className` to the end, as an optional property, so it can be omitted in the 90% of the calls you have below?